src/pgduckdb_types.cpp

```diff
@@ -1265,12 +1267,33 @@ ConvertPostgresToBaseDuckColumnType(Form_pg_attribute &attribute) {
 		return duckdb::LogicalTypeId::DOUBLE;
 	case NUMERICOID:
 	case NUMERICARRAYOID: {
-		auto &typmod = attribute->atttypmod;
-		auto precision = numeric_typmod_precision(typmod);
-		auto scale = numeric_typmod_scale(typmod);
-		if (typmod == -1 || precision < 0 || scale < 0 || precision > 38) {
-			auto extra_type_info = duckdb::make_shared_ptr<NumericAsDouble>();
-			return duckdb::LogicalType(duckdb::LogicalTypeId::DOUBLE, std::move(extra_type_info));
+		auto precision = numeric_typmod_precision(type_modifier);
+		auto scale = numeric_typmod_scale(type_modifier);
+		if (type_modifier == -1 || precision < 1 || precision > 38 || scale < 0 || scale > 38 || scale > precision) {
```

Comment:
Would be great to replace 38 with:
```
std::numeric_limits<float>::max_exponent10
```

(also is `duckdb::LogicalTypeId::DOUBLE` backed by `float` not a `double`? that sounds surprising?)

Reply:
the 38 has nothing to do with floats, it's because the maximum number of decimal digits that fit into an `int128` is 38: https://duckdb.org/docs/stable/sql/data_types/numeric.html#fixed-point-decimals

Reply:
Ah yeah that makes more sense :) Is there a DuckDB constant we can use?

Reply:
Maybe here https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/common/limits.hpp#L56

Reply:
Given that the number there is 39 and the docs use 38, I decided to simply refer to the docs in a code comment. Also if numerics ever get more precision using the constant you linked wouldn't help in updating it automatically, since that is linked specifically to `hugeint_t`